var/spack/repos/builtin/packages/soapdenovo-trans/package.py

```diff
@@ -0,0 +1,47 @@
+##############################################################################
+# Copyright (c) 2013-2018, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class SoapdenovoTrans(Package):
+    """SOAPdenovo-Trans is a de novo transcriptome assembler basing on the
+       SOAPdenovo framework, adapt to alternative splicing and different
+       expression level among transcripts."""
+
+    homepage = "http://soap.genomics.org.cn/SOAPdenovo-Trans.html"
+    url      = "https://github.com/aquaskyline/SOAPdenovo-Trans/archive/1.0.4.tar.gz"
+
+    version('1.0.4', 'a3b00b0f743b96141c4d5f1b49f2918c')
+
+    def install(self, spec, prefix):
+        makefile = './src/Makefile'
+        filter_file(r'CFLAGS=         -O3 -fomit-frame-pointer -static',
+                            'CFLAGS=         -O3 -fomit-frame-pointer',
+                            makefile)
+
+        sh = which('sh')
+        sh('./make.sh')
```

Comment:
It looks like `make.sh` just runs:
```bash
cd src/
make
make 127mer=1
```
I would rather this be written as a `MakefilePackage` where we set:
```python
build_directory = 'src'
```
We can edit the Makefile in the `edit` phase and run `make` twice in the `build` phase. Let me know if you have any questions. I can help with this if you need it.

Reply:
I tried this, but it seems that '127mer=1' is not a make target, so when I run the second make('127mer=1') in build phase, spack doesn't recognize it and report some error like this:

==> Error: ProcessError: Command exited with status 2:
    'make' '-j24' '127mer=1'

47 errors found in build log:
     42    scaffold.c:73:4: warning: implicit declaration of function ‘RPKMStat’ [-Wimplicit-function-declaration]
     43        RPKMStat(graphfile);
     44        ^~~~~~~~
../SOAPdenovo-Trans-31mer compilation done.
     46    ==> 'make' '-j24' '127mer=1'
     47    ../SOAPdenovo-Trans-127mer cleaning done.
  >> 48    gcc: error: arc.o: No such file or directory
  >> 49    gcc: error: attachPEinfo.o: No such file or directory
  >> 50    gcc: error: bubble.o: No such file or directory

Have no idea what's going on.

Reply:
What does your build phase look like?

Reply:
never mind, I set parallel=False and run again, it works now.